src/xtgeo/xyz/_xyz.py

```diff
@@ -3,48 +3,115 @@
 from collections import OrderedDict
 from copy import deepcopy
 import pathlib
+import functools
+import warnings
+import deprecation
+import io
+from typing import Union, Optional, Any
 
 import pandas as pd
 
 import xtgeo
 from xtgeo.common import XTGeoDialog, XTGDescription
 from xtgeo.xyz import _xyz_io
 from xtgeo.xyz import _xyz_roxapi
+import xtgeo.common.sys as xtgeosys
 
 xtg = XTGeoDialog()
 logger = xtg.functionlogger(__name__)
 
 
+def _data_reader_factory(file_format):
+    if file_format in ("xyz", "poi", "pol"):
+        return _xyz_io.import_xyz
+    if file_format == "zmap":
+        return _xyz_io.import_zmap
+    if file_format in ("rms_attr", "rmsattr"):
+        return _xyz_io.import_rms_attr
+    raise ValueError(f"Unknown file format {file_format}")
+
+
+def allow_deprecated_init(func):
+    # This decorator is here to maintain backwards compatibility in the construction
+    # of Points/Polygons and should be deleted once the deprecation period has expired,
+    # the construction will then follow the new pattern.
+    @functools.wraps(func)
+    def wrapper(cls, spec, fformat="xyz", zname="Z_TVDSS", is_polygons=False):
+        # Checking if we are doing an initialization from file and raise a
+        # deprecation warning if we are.
+        derived_importname = "points_from_file"
+        if is_polygons:
```

Comment:
Instead of adding another property to the object in order to have it show up in the wrapper here, should probably move and/or copy `allow_deprecated_init` to implementing classes `Points` and `Polygons` so we can have explicit warning messages.

Reply:
I see the error message was changed later, but the addition of the `is_polygons` parameter should probably not be in the commit dc71df70eb98c76a600abf4f9a28c89e754f228e which relates to the #524 .